usecases/memwatch/monitor_test.go

```diff
@@ -159,37 +159,43 @@ func TestMappings(t *testing.T) {
 		mappingsLeft := getMaxMemoryMappings() - currentMappings
 		assert.InDelta(t, mappingsLeft, addMappings, 10) // other things can happen at the same time
 		path := t.TempDir()
+
+		limitReached := false
+
 		// use up available mappings
-		for i := 0; i < int(mappingsLeft)+5; i++ {
+		for i := 0; i < int(mappingsLeft)*2; i++ {
 			m.Refresh(true)
 			file, err := os.OpenFile(path+"example"+strconv.FormatInt(int64(i), 10)+".txt", os.O_CREATE|os.O_RDWR, 0o666)
 			require.Nil(t, err)
-			defer file.Close()
+			defer file.Close() // files should stay open until end of test to continue to use mappings
```

Comment:
**suggestion (if-minor):** could we sate explicitly a note that defer in loop it's made here in purpose in order to keep all files open until the for loop finishes

Reply:
i wanna make sure, that anyone will come through that test don't spend time thinking the defer in the loop is the issue, but that it's made in purpose to keep resources open given that we gonna go that way